core/trino-spi/src/main/java/io/trino/spi/connector/WriterScalingOptions.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public record WriterScalingOptions(boolean isWriterTasksScalingEnabled, boolean isPerTaskWriterScalingEnabled, Optional<Integer> taskScalingMaxWriterCount)
```

Comment:
`taskScalingMaxWriterCount` - is this per task or per table writer node?

Reply:
I'm not clear on why `getMaxWriterTasks` was introduced. If it was to limit the maximum amount of write parallelism to an external data source, then the presence of per-task scaling already breaks that.
Per task writer scaling should probably be disabled when `getMaxWriterTasks` returns something non-empty, as it appears that jdbc connectors call this "write_parallelism".
We should probably deal with that separately to avoid getting blocked on that.

Reply:
+1

Reply:
`getMaxWriterTasks` was introduced to limit number of tasks. From what I know `jdbc` connectors need that and use that - I know it from talks with different people - I did not confirm that so I am not sure. 
It does not interfere with parallelism within task because it limit statically number of tasks.

Reply:
I can see that getMaxWriterTasks limits number of writing tasks. The question is what underlying problem were we trying to solve by doing that ? If this was about limiting parallelism, then we need to change `getMaxWriterTasks` to `getMaxWriterParallelism` and implementing that should disable within task writer scaling. We can do it as a follow-up since landing this PR will side step the problem for now, but we should fix the API.

Reply:
> We can do it as a follow-up since landing this PR will side step the problem for now, but we should fix the API.

Yes, we should to it.